### PR TITLE
feat: External navigation nodes as part of extensibility

### PR DESCRIPTION
**Pull request by OliwiaGowor**
**Description**

Changes proposed in this pull request:

- added external navigation nodes to extensibility 
- added ability to change scope for the busola-config external nodes
- adjusted documentation

**Related issue(s)**
#2887 

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
```
  externalNodes:
    - category: TEST123
      children:
        - label: TEST_LABEL
          link: 'http://github.com/kyma-project/lifecycle-manager/blob/main/docs/technical-reference/api/kyma-cr.md'
```
NavItem:73
image

File: docs/extensibility/resources.md [hunk 1]
```diff
@@ -45,6 +45,13 @@ The **general** section is required and contains basic information about the res
     - **disableCreate** - when set to `true`, it disables the **Create** button. Defaults to `false`.
     - **disableEdit** - when set to `true`, it disables the **Edit** button. Defaults to `false`.
     - **disableDelete** - when set to `true`, it disables the **Delete** button. Defaults to `false`.
+- **externalNodes** - a optional list of links to external websites.
+  - **category** - a category name
+  - **scope** - either `namespace` or `cluster`. Defaults to `cluster`.
+  - **icon** - an optional icon
```
Comment (mrCherry97) [inline docs/extensibility/resources.md:51]: We need to provide an information where user can find a list of the available icons

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Icons are not working 
```
  externalNodes:
    - category: TEST123
      scope: namespace
      children:
        - label: TEST_LABEL_KYMA1
          link: '$string(($apiRules().items.spec.host)[0])'
          icon: accidental-leave
        - icon: competitor
          label: TEST_LABEL_KYMA2
          link: 'www.google.com'
```
image

File: src/sidebar/NavItem.tsx [hunk 2]
```diff
@@ -36,6 +39,14 @@ export function NavItem({ node, subItem = false }: NavItemProps) {
   const namespaceId = useRecoilValue(activeNamespaceIdState);
   const cluster = useRecoilValue(clusterState);
 
+  const jsonata = useJsonata({ resource: {} as Resource });
+  const [jsonataLink, setJsonataLink] = useState('');
+
+  useEffect(() => {
+    const [jsonataLinkTmp] = jsonata(node.externalUrl || '');
+    setJsonataLink(jsonataLinkTmp);
+  }, [jsonata, node.externalUrl]);
```
Comment (mrCherry97) [inline src/sidebar/NavItem.tsx:48]: Infinite loop, remove useEffect

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

File: src/components/Extensibility/contexts/DataSources.tsx [hunk 3]
```diff
@@ -92,7 +92,7 @@ export const DataSourcesContextProvider: FC<Props> = ({
       resource: { group, kind, version, name, namespace },
       ownerLabelSelectorPath,
     } = dataSource;
-    if (typeof namespace === 'undefined') {
+    if (typeof namespace === 'undefined' && resource.metadata) {
       namespace = resource.metadata.namespace;
```
Comment (mrCherry97) [inline src/components/Extensibility/contexts/DataSources.tsx:96]:
```suggestion
    if (typeof namespace === 'undefined') {
      namespace = resource?.metadata?.namespace;
```

File: src/sidebar/NavItem.tsx [hunk 4]
```diff
@@ -56,12 +67,13 @@ export function NavItem({ node, subItem = false }: NavItemProps) {
...
     onClick: (e: Event) => {
-      if (node.externalUrl) {
-        const newWindow = window.open(
-          node.externalUrl,
-          '_blank',
-          'noopener, noreferrer',
-        );
+      if (node.dataSources) {
+        let link = jsonataLink || node.externalUrl || '';
+        link = link.startsWith('http') ? link : `https://${link}`;
```
Comment (mrCherry97) [inline src/sidebar/NavItem.tsx:72]: it's not working

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

**Review by mrCherry97 — APPROVED** [review]

File: docs/extensibility/resources.md [hunk 5]
```diff
@@ -63,6 +70,12 @@ features:
   actions:
     disableCreate: true
     disableDelete: true
+externalNodes:
+  - category: My Category
+    icon: course-book
+    children:
+      - label: Example Node Label
+        link: 'https://github.com/kyma-project/busola' & $string($exampleResource().link)
```
Comment (mrCherry97) [inline docs/extensibility/resources.md:78]:
```suggestion
externalNodes:
  - category: My Category
    icon: course-book
    children:
      - label: Example Node Label
        link: 'https://github.com/kyma-project/busola'
  - category: My Second Category
    icon: bar-chart
    scope: namespace
    children:
      - label: Example Node Label
        link: '$string($exampleResource().link)'
```

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

File: docs/extensibility/resources.md [hunk 6]
```diff
@@ -45,6 +45,13 @@ The **general** section is required and contains basic information about the res
     - **disableCreate** - when set to `true`, it disables the **Create** button. Defaults to `false`.
     - **disableEdit** - when set to `true`, it disables the **Edit** button. Defaults to `false`.
     - **disableDelete** - when set to `true`, it disables the **Delete** button. Defaults to `false`.
+- **externalNodes** - a optional list of links to external websites.
```
Comment (grego952) [inline docs/extensibility/resources.md:48]:
```suggestion
- **externalNodes** - an optional list of links to external websites.
```

File: docs/extensibility/resources.md [hunk 7]
```diff
@@ -45,6 +45,13 @@ The **general** section is required and contains basic information about the res
...
+  - **category** - a category name
+  - **scope** - either `namespace` or `cluster`. Defaults to `cluster`.
+  - **icon** - an optional icon. You can find the list of available icons [here](https://sdk.openui5.org/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview)
+  - **children** - a list of child nodes containing details about the links
+    - **label** - a displayed label
+    - **link** - a link to external website. A [JSONata](jsonata.md) function can be provided.
```
Comment (grego952) [inline docs/extensibility/resources.md:54]:
```suggestion
    - **link** - a link to an external website. You can provide a [JSONata](jsonata.md) function.
```

File: docs/extensibility/resources.md [hunk 8]
```diff
@@ -45,6 +45,13 @@ The **general** section is required and contains basic information about the res
     - **disableCreate** - when set to `true`, it disables the **Create** button. Defaults to `false`.
     - **disableEdit** - when set to `true`, it disables the **Edit** button. Defaults to `false`.
     - **disableDelete** - when set to `true`, it disables the **Delete** button. Defaults to `false`.
+- **externalNodes** - a optional list of links to external websites.
+  - **category** - a category name
+  - **scope** - either `namespace` or `cluster`. Defaults to `cluster`.
+  - **icon** - an optional icon. You can find the list of available icons [here](https://sdk.openui5.org/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview)
```
Comment (grego952) [inline docs/extensibility/resources.md:51]:
```suggestion
  - **icon** - an optional icon. Go to [Icon Explorer](https://sdk.openui5.org/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview) to find a list of the available icons.
```

**Review by grego952 — APPROVED** [review]

**Review by mrCherry97 — APPROVED** [review]

**Pull request merged**